examples/fuzzer.rs

```diff
@@ -813,6 +803,132 @@ fn record_graphic_msaa_depth_stencil(frame: &mut FrameContext, pool: &mut HashPo
         });
 }
 
+fn record_graphic_will_merge_common_color(frame: &mut FrameContext, pool: &mut HashPool) {
+    let pipeline = graphic_vert_frag_pipeline(
+        frame.device,
+        GraphicPipelineInfo::default(),
+        inline_spirv!(
+            r#"
+            #version 460 core
+
+            void main() {
+            }
+            "#,
+            vert
+        )
+        .as_slice(),
+        inline_spirv!(
+            r#"
+            #version 460 core
+
+            layout(location = 0) out vec4 color_out;
+
+            void main() {
+                color_out = vec4(0);
+            }
+            "#,
+            frag
+        )
+        .as_slice(),
+    );
+    let image = frame.render_graph.bind_node(
+        pool.lease(ImageInfo::image_2d(
+            256,
+            256,
+            vk::Format::R8G8B8A8_UNORM,
+            vk::ImageUsageFlags::COLOR_ATTACHMENT,
+        ))
+        .unwrap(),
+    );
+
+    // Pass "a" stores color0 which "b" compatibly loads; so these two will get merged
+    frame
+        .render_graph
+        .begin_pass("a")
+        .bind_pipeline(&pipeline)
+        .store_color(0, image)
+        .record_subpass(|subpass, _| {
+            subpass.draw(1, 1, 0, 0);
+        });
+    frame
+        .render_graph
+        .begin_pass("b")
+        .bind_pipeline(&pipeline)
+        .load_color(0, image)
+        .store_color(0, image)
+        .record_subpass(|subpass, _| {
+            subpass.draw(1, 1, 0, 0);
+        });
+}
+
+fn record_graphic_will_merge_common_depth1(frame: &mut FrameContext, pool: &mut HashPool) {
+    let pipeline = graphic_vert_frag_pipeline(
+        frame.device,
+        GraphicPipelineInfo::default(),
+        inline_spirv!(
+            r#"
+            #version 460 core
+
+            void main() {
+            }
+            "#,
+            vert
+        )
+        .as_slice(),
+        inline_spirv!(
+            r#"
+            #version 460 core
+
+            layout(location = 0) out vec4 color_out;
+
+            void main() {
+                color_out = vec4(0);
+            }
+            "#,
+            frag
+        )
+        .as_slice(),
+    );
+    let color_image = frame.render_graph.bind_node(
+        pool.lease(ImageInfo::image_2d(
+            256,
+            256,
+            vk::Format::R8G8B8A8_UNORM,
+            vk::ImageUsageFlags::COLOR_ATTACHMENT,
+        ))
+        .unwrap(),
+    );
+    let depth_image = frame.render_graph.bind_node(
+        pool.lease(ImageInfo::image_2d(
+            256,
+            256,
+            vk::Format::D32_SFLOAT,
+            vk::ImageUsageFlags::DEPTH_STENCIL_ATTACHMENT,
+        ))
+        .unwrap(),
+    );
+
+    // Pass "a" stores color0+depth which "b" compatibly loads; so these two will get merged
+    frame
+        .render_graph
```

Comment:
Here is an example of "Color0+Depth" merging with "Depth"

Reply:
Other examples are needed because I think the latest commit accidentally broke Depth merging with Depth, which is an odd use case but still valid.

Reply:
Fixed ^^